menu/src/main/java/com/jagrosh/jdautilities/menu/Menu.java

```diff
@@ -60,14 +62,19 @@
     protected Set<Role> roles;
     protected final long timeout;
     protected final TimeUnit unit;
-    
-    protected Menu(EventWaiter waiter, Set<User> users, Set<Role> roles, long timeout, TimeUnit unit)
+
+    private final Consumer<Message> finalAction;
+    private Future<Void> cancelFuture;
+    private Message attachedMessage;
+
+    protected Menu(EventWaiter waiter, Set<User> users, Set<Role> roles, long timeout, TimeUnit unit, Consumer<Message> finalAction)
```

Comment:
Unfortunately, this breaks compatibility with previous versions for people who implemented their own menus.